staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy.go

```diff
@@ -33,28 +33,22 @@ func NewStatusStrategy(strategy customResourceStrategy) statusStrategy {
 }
 
 func (a statusStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	// update is only allowed to set status
 	newCustomResourceObject := obj.(*unstructured.Unstructured)
-	oldCustomResourceObject := old.(*unstructured.Unstructured)
-
 	newCustomResource := newCustomResourceObject.UnstructuredContent()
-	oldCustomResource := oldCustomResourceObject.UnstructuredContent()
-
-	// update is not allowed to set spec and metadata
-	_, ok1 := newCustomResource["spec"]
-	_, ok2 := oldCustomResource["spec"]
-	switch {
-	case ok2:
-		newCustomResource["spec"] = oldCustomResource["spec"]
-	case ok1:
-		delete(newCustomResource, "spec")
-	}
+	status, ok := newCustomResource["status"]
 
-	newCustomResourceObject.SetAnnotations(oldCustomResourceObject.GetAnnotations())
-	newCustomResourceObject.SetFinalizers(oldCustomResourceObject.GetFinalizers())
```

Comment:
Not related to this PR, but interesting nonetheless: 

Wondering whether this finalizers override means that a controller that only may write to /status is stopped from adding/removing itself to the finalizers. Is this intended? /cc @nikhita @deads2k @liggitt

Reply:
Similar question about annotations.

Reply:
> Wondering whether this finalizers override means that a controller that only may write to /status is stopped from adding/removing itself to the finalizers. Is this intended? /cc @nikhita @deads2k @liggitt

Mid-term we need a /finalize subresource.  In such a world the finalizers are restricted.  I think tightening it here is ok.  Finalization controllers are special.

Reply:
> I think tightening it here is ok

Which means we break finalizers for people as soon as they enable /status. I don't like that.

Reply:
> Which means we break finalizers for people as soon as they enable /status. I don't like that.

They have got to some effort to use the status endpoint, so they don't get here by accident.  Updating status isn't a license to change metadata.  Finalization controls the deletion of an API type, it's not a status field.